pyomo/contrib/mindtpy/util.py

```diff
@@ -421,40 +419,72 @@ def set_solver_options(opt, solve_data, config, type, regularization=False):
     # nlp_args = dict(config.nlp_solver_args)
     elapsed = get_main_elapsed_time(solve_data.timing)
     remaining = int(max(config.time_limit - elapsed, 1))
-    if type == 'mip':
+    if solver_type == 'mip':
         solver_name = config.mip_solver
         if regularization:
             if config.projection_mip_threads > 0:
                 opt.options['threads'] = config.projection_mip_threads
         else:
             if config.threads > 0:
                 opt.options['threads'] = config.threads
-    elif type == 'nlp':
+    elif solver_type == 'nlp':
         solver_name = config.nlp_solver
     # TODO: opt.name doesn't work for GAMS
     if solver_name in {'cplex', 'gurobi', 'gurobi_persistent'}:
         opt.options['timelimit'] = remaining
         opt.options['mipgap'] = 0.001
         if regularization == True:
             if solver_name == 'cplex':
-                opt.options['mip limits populate'] == 10
+                opt.options['mip limits populate'] = 10
```

Comment:
What `optcr` value should we set for NLP solvers?

Reply:
I do not care too much about the optcr for NLP solvers, while I do care for the MIP. This since for example in GAMS that value is by default 0.1. We do have optimality tolerances in MindtPy, so we can apply those same tolerances for the subproblems.